### ISSUE TITLE
[BUG]: Microsoft Advertising app removal failure

### ISSUE THREAD

**Issue by djacu** [undergroundwires/privacy.sexy#200]
### Description

Microsoft Advertising app fails to remove using the standard script.

### OS

Windows 10 Pro
Version 22H2
Build 19045.2604

### Reproduction steps

Download Standard script for Windows
Right click file and run as administrator

### Scripts

[privacy-script.txt](https://github.com/undergroundwires/privacy.sexy/files/10780444/privacy-script.txt)

### Screenshots

```
--- Microsoft Advertising app

Remove-AppxPackage: Deployment failed with HRESULT: 0x80073CF3, Package failed updates, dependency or conflict validation. Windows cannot remove framework Microsoft. Advertising. Xaml_10.1808.3.0_x64_8wekyb3d8bbwe because package(s) microsoft.windows communicationsapps currently depends on the framework. Removing all packages that depend on the framework automatically removes the framework.

NOTE: For additional information, look for [ActivityId] 4dd46208-44cf-0004-3e70-d44dcf44d901 in the Event Log or use the command line Get-AppPackageLog -ActivityID 4dd46208-44cf-0004-3e70-d44dcf44d901

At line:1 char:48

+Get-AppxPackage 'Microsoft.Advertising. Xaml' | Remove-AppxPackage

+ : WriteError: (Microsoft.Adver..._8wekyb3d8bbwe: String) [Remove-AppxPackage], IOException

+ CategoryInfo

+ FullyQualifiedErrorId: DeploymentError, Microsoft.Windows. Appx.PackageManager.Commands. RemoveAppxPackageCommand
```

**Comment by undergroundwires** [comment]
Hi @djacu, thanks for the bug report. I tested this and can reproduce this.

**Windows 11 22H2**: Works fine, because `Microsoft.Advertising.Xaml` does not exist.
**Windows 10 21H2**: It prompts me the delete (MSN weather app) `Microsoft.BingWeather` and (Mail and Calendar app) `microsoft.windowscommunicationsapps` apps first. Both can be deleted by privacy.sexy. I get this message:

```sh
Windows cannot remove framework Microsoft.Advertising.Xaml_10.1808.3.0_x64__8wekyb3d8bbwe because package(s) microsoft.windowscommunicationsapps Microsoft.BingWeather currently depends on the framework.
Removing all packages that depend on the framework automatically removes the framework.
```

My suggestion is actually remove this script completely from privacy.sexy due to:

- Being unsupported on Windows 11.
- Being automatically deleted once the dependent packages (bing weather and communications apps) are deleted. I tested that and it's actually removed after removing these packages:
  1. Remove "Mail and calendar app": `PowerShell -ExecutionPolicy Unrestricted -Command "Get-AppxPackage 'microsoft.windowscommunicationsapps' | Remove-AppxPackage"`
  2. Remove "MSN Weather app": ``PowerShell -ExecutionPolicy Unrestricted -Command "Get-AppxPackage 'microsoft.windowscommunicationsapps' | Remove-AppxPackage"`
  3. Check whether Microsoft Advertising app exists: `Get-AppxPackage 'Microsoft.Advertising.Xaml'` (should output empty)

Another option would be doing a soft delete of `Microsoft.Advertising.Xaml` by moving its folder (breaking it) but it's probably not worth it as it gets cleaned up after its upstream dependencies are deleted. It could also have unintended side-effects on upstream dependencies.

Sorry for late response, let me know if you're around and this make sense @djacu.
